### PR TITLE
OSSM-8252: Add pod locality controller fails to update pod test

### PR DESCRIPTION
**Pull request by unsortedhashsets**
Related (test covers two issues):
- [OSSM-2169](https://issues.redhat.com/browse/OSSM-2169)
- [OSSM-2420](https://issues.redhat.com/browse/OSSM-2420)
- MTT run with GA operator: [3061](https://master-jenkins-csb-servicemesh.apps.ocp-c1.prod.psi.redhat.com/job/maistra/job/maistra-test-tool/3061/)

### CODE REVIEW

File: pkg/tests/ossm/bug_istiopods_test.go [hunk 1]
```diff
@@ -56,6 +60,72 @@ func TestIstiodPodFailsAfterRestarts(t *testing.T) {
...
+			oc.DeleteNamespace(t, namespaces...)
+			oc.ApplyString(t, meshNamespace, smmr)
+		})
+
+		DeployControlPlane(t)
+
+		t.LogStep("Add namespaces to the SMMR")
+		oc.ApplyString(t, meshNamespace, createSMMRManifest(namespaces...))
+
+		istioOperatorPodName := oc.GetResouceNameByLabel(t, "openshift-operators", "pod", "name=istio-operator")
```
Comment (unsortedhashsets) [inline pkg/tests/ossm/bug_istiopods_test.go:88]: I will replace it after merge 755

File: pkg/tests/ossm/bug_istiopods_test.go [hunk 2]
```diff
@@ -56,6 +60,72 @@ func TestIstiodPodFailsAfterRestarts(t *testing.T) {
...
+				} else {
+					t.Log(output)
+					t.Fatalf("Was not found success message after error for namespace %s: %s", namespaceName, successMessage)
+				}
+			} else {
+				if count < 6 {
+					count++
+					t.Logf("Was not found any 'error adding member-of label' error, repeat (max 5), attempt #%d", i)
+				} else {
+					t.Logf("Was not found any 'error adding member-of label' error, stop test, attempt #%d", i)
```
Comment (mkralik3) [inline pkg/tests/ossm/bug_istiopods_test.go:122]: `stop test` , shouldn't be there t.Fail or something?

Comment (unsortedhashsets) [reply pkg/tests/ossm/bug_istiopods_test.go:122]: I decided to make just successful stop after 5 attempts to face error.

I found out one new issue with latest mtt run so will try to rework that part too

Comment (unsortedhashsets) [reply pkg/tests/ossm/bug_istiopods_test.go:122]: As it is flaky conflict that previously led to error... So I do not know should we spent extra time to always trying to catch that conflict :thinking:

**Review by mkralik3 — APPROVED** [review]

**Pull request merged**